[CI] StatelessFileDeletionIT testDeleteIndexAfterRecovery failing

**Build Scans:**
- [elasticsearch-intake #40724 / part2](https://gradle-enterprise.elastic.co/s/c44ml7ho7ki4c)
- [elasticsearch-pull-request #151104 / part-2](https://gradle-enterprise.elastic.co/s/hk2ibwvigy6ow)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:stateless:internalClusterTest" --tests "org.elasticsearch.xpack.stateless.commits.StatelessFileDeletionIT.testDeleteIndexAfterRecovery" -Dtests.seed=3B49136DD6ABE346 -Dtests.locale=kde -Dtests.timezone=Asia/Jayapura -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.stateless.commits.StatelessFileDeletionIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testDeleteIndexAfterRecovery'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: an empty collection
     but: <[indices/yxD8fflAThSkpwuPYPU8uA/0/1/stateless_commit_3]>
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testDeleteIndexAfterRecovery
- [main] 1 consecutive failures in step part2
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 2 failures in test testDeleteIndexAfterRecovery (0.5% fail rate in 371 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

```
java.lang.AssertionError: |  
-- | --
Expected: an empty collection |  
but: <[indices/Jf09wjtQTOyaeRGlcxCbYw/0/1/stateless_commit_3]> |  

at __randomizedtesting.SeedInfo.seed([6391C23B8EAA9C54:9991B6FBA57D30F9]:0)	
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)	
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)	
	at org.elasticsearch.test.ESTestCase.assertThat(ESTestCase.java:2980)	
	at org.elasticsearch.xpack.stateless.commits.StatelessFileDeletionIT.lambda$testDeleteIndexAfterRecovery$29(StatelessFileDeletionIT.java:1074)	
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1724)	
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1696)	
	at org.elasticsearch.xpack.stateless.commits.StatelessFileDeletionIT.testDeleteIndexAfterRecovery(StatelessFileDeletionIT.java:1074)	
```

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -455,9 +455,6 @@ tests:
 - class: org.elasticsearch.compute.aggregation.SumLongAggregatorFunctionTests
   method: testOverflowFails
   issue: https://github.com/elastic/elasticsearch/issues/150455
-- class: org.elasticsearch.xpack.stateless.commits.StatelessFileDeletionIT
-  method: testDeleteIndexAfterRecovery
-  issue: https://github.com/elastic/elasticsearch/issues/150474
 - class: org.elasticsearch.xpack.esql.analysis.AnalyzerUnmappedGoldenTests
   method: testTypeConflictTimeseriesStatsWithCast
   issue: https://github.com/elastic/elasticsearch/issues/150476

--- a/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/commits/StatelessFileDeletionIT.java
+++ b/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/commits/StatelessFileDeletionIT.java
@@ -97,6 +97,7 @@ import org.elasticsearch.xpack.stateless.lucene.BlobStoreCacheDirectory;
 import org.elasticsearch.xpack.stateless.lucene.IndexBlobStoreCacheDirectory;
 import org.elasticsearch.xpack.stateless.objectstore.ObjectStoreService;
 import org.elasticsearch.xpack.stateless.objectstore.ObjectStoreTestUtils;
+import org.elasticsearch.xpack.stateless.objectstore.gc.ObjectStoreGCTask;
 import org.elasticsearch.xpack.stateless.recovery.TransportRegisterCommitForRecoveryAction;
 
 import java.io.IOException;
@@ -1045,7 +1046,12 @@ public class StatelessFileDeletionIT extends AbstractStatelessPluginIntegTestCas
 
     public void testDeleteIndexAfterRecovery() throws Exception {
         startMasterOnlyNode();
-        final var indexNodeA = startIndexNode();
+        // This test exposes a race between index deletion and relocation described in #150474.
+        // In production commit files are eventually deleted by `StaleIndicesGCService` so we enable it in the test too.
+        final var indexNodeSettings = nodeSettings().put(ObjectStoreGCTask.GC_INTERVAL_SETTING.getKey(), TimeValue.timeValueSeconds(1))
+            .build();
+
+        final var indexNodeA = startIndexNode(indexNodeSettings);
         final var searchNodeA = startSearchNode();
         var indexName = randomIdentifier();
         createIndex(indexName, 1, 1);
@@ -1053,7 +1059,7 @@ public class StatelessFileDeletionIT extends AbstractStatelessPluginIntegTestCas
         var totalIndexedDocs = indexDocsAndFlush(indexName);
         var shardCommitsContainer = getShardCommitsContainerForCurrentPrimaryTerm(indexName, indexNodeA, 0);
 
-        startIndexNode();
+        startIndexNode(indexNodeSettings);
         startSearchNode();
         ensureStableCluster(5);
         final var excludeIndexOrSearchNode = randomBoolean();

